jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/AsyncContent.java

```diff
@@ -79,7 +79,7 @@ public void write(boolean last, ByteBuffer byteBuffer, Callback callback)
      * or succeeded if and only if the chunk is terminal, as non-terminal
      * chunks have to bind the succeeding of the callback to their release.
      */
-    private void offer(Content.Chunk chunk)
+    protected void offer(Content.Chunk chunk)
```

Comment:
I remember @lorban and I discussing whether to expose this method, and IIRC we made it `private` because we did not want users to be able to add other implementations of `Chunk`, mess with the `last`-ness and mess with the reference counting.

The implementation seems to have some relic of the historical implementation, when all types of chunk were allowed and can be cleaned up.

I would make it `private` again, as it is only used in a couple of tests.